### ISSUE TITLE
Create a way to map a Locale to a default Currency

### ISSUE THREAD

**Issue by blaynem** [unicode-org/icu4x#5316]
## Problem Statement

I want to format a Currency for a given locale. I must now find and provide both the `locale` and the `currencyCode` ("USD", "EUR", etc) in order to format. A good example is in the currency [CurrencyFormatter](https://github.com/unicode-org/icu4x/blob/main/components/experimental/src/dimension/currency/formatter.rs#L131-L134), or even in the JS [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) for currency.

What if my list of Currency Codes is misaligned from the formatter data? Wouldn't it be nice if I could get all the data I needed from the same exact source?

## Questions

1. Is this something that should even live in ICU4X? If so, where (Locale, Currency, etc)? 
2. Suggestions on getting started?

## Useful Docs
- [Territory Currency Info](https://www.unicode.org/cldr/charts/45/supplemental/detailed_territory_currency_information.html)
- In [Supplemental Currency Data](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/currencyData.json#L337) the `currencyData.region` is an array of Currency codes and when they were used.

**Comment by sffc** [comment]
The primary use case behind wanting to map a region to a currency is because you want to show prices using that currency. However, you cannot simply "show prices" in an arbitrary currency; you need to either convert the prices with a constantly-fluctuating exchange rate, or you need to pull pre-calculated prices from your sales department (which may incorporate factors such as cost of living, freight, taxes/fees, etc).

The ICU4X and Intl currency formatters are designed to support any arbitrary currencies. Countries change or add currencies all the time, but ICU4X will happily format in whichever currency provided to it.

**Comment by blaynem** [comment]
I might need to clarify! I don't expect the Currency Formatter to do any of the exchange rates.

For us, a client would likely be passed something like below.  If our `currencyCode` goes out of sync with icu4x's, then we'll get some errors.
```js
const returned = {
  "localization": { "locale": "en-US" },
  "body": {
    "widget": "currency",
    "value": "1234567.89",
    "currencyCode": "USD", // If this changes to USB
    "sign": "$"
  }
}
```

My proposal would be more of a way to provide a locale and get back the Currency code. 
```rs
pub struct CurrencyCode(pub TinyAsciiStr<3>);

impl CurrencyCode {
    fn create(locale: &DataLocale) -> Result<Self, DataError> {
        // Do magic, get the CurrencyCode for the given locale
        let found_code = do_magic();
        Ok(Self (found_code)) // CurrencyCode("USD")
    }
}
```

Though after writing all this out, it feels more like a random field we should add into our own data / repo... 😅

**Comment by zbraniecki** [comment]
I think your example still omits the problem that Shane raised. 

```
const returned = {
  "localization": { "locale": "en-US" },
  "body": {
    "widget": "currency",
    "value": "1234567.89",
    "currencyCode": "USD", // If this changes to USB
    "sign": "$"
  }
}
```

This data encodes that something has 1234567.89 united state dollars. The locale doesn't matter for the unit of the quantity (quantity defined as `{value, unit}`).

You may reasonably ask "how to format 1234567.89 united states dollars to display it in a given locale" because that can be displayed differently in USA, from how it would be displayed in Canada, China, France or Australia.

But you cannot do what I interpret from your request - you can't give ICU4X a locale (en-US) and a value (1234567.89) and ask it to give you back the correct currency unit for that value and locale.

That being said, there are reasons to retrieve "common currencies for a locale" (I would advise against assuming there is only one).
For example, knowing a user locale you may want to display common currencies for that locale at the top of currency selector. But I don't think you need ICU4X that - it's not an i18n operation. It's just a database of locale -> common currencies.

**Comment by blaynem** [comment]
Yeah I think my entire request basically boils down to your last sentence haha

> it's not an i18n operation. It's just a database of locale -> common currencies.

Thanks!

**Comment by sffc** [comment]
"locale to list of common currencies" is approximately as valid as "locale to list of common time zones". I'd say that's not _out of scope_, but it's not as much of a core operation

**Comment by blaynem** [comment]
For anyone who comes along and wonders how they might do this in the future!

The simplest way is if you have a region, you can make a "best guess" what the default Currency might be with the [CLDR Supplemental CurrencyData](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/currencyData.json#L337).

Our solution is to assume if there are either `_tender` or `_to` fields present, it's probably not the default. It worked for our ~85 locales, but can't promise it works for everything.
